packages/textcomplete-core/src/SearchResult.ts

```diff
@@ -22,11 +26,24 @@ export class SearchResult<T = unknown> {
     const replacement = result
       .replace(MAIN, match[0])
       .replace(PLACE, (_, p) => match[parseInt(p)])
+
+    return {
+      start: match.index,
+      end: match.index + match[0].length,
+      text: replacement,
+    }
+  }
+
+  replace(beforeCursor: string, afterCursor: string): [string, string] | void {
+    const replacement = this.getReplacementData(beforeCursor, afterCursor)
+
+    if (replacement === undefined) return
```

Comment:
nits: You should use the same expression to check returned value to keep a consistency with https://github.com/increments/textcomplete/pull/1/files#diff-0772563b5ba63266e41fd9281c4aac7d9fe86b827484af9d4eb61c190f473c05R25 .